tests/Microsoft.Identity.Test.LabInfrastructure/Microsoft.Identity.Test.LabInfrastructure.csproj

```diff
@@ -5,13 +5,17 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.KeyVault" Version="3.0.2" />
-    <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="4.4.2" />
+    <PackageReference Include="Microsoft.Azure.KeyVault" Version="3.0.3" />
+    <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="4.5.0" />
```

Comment:
><PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="4.5.0" /> [](start = 4, length = 94)

we should really move this to use MSAL

Reply:
MSAL can get tokens which are for a v1 SDK. The AT will likely be v1 (but as a developer you shouldn't have to worry about that) and the service will be able to use existing code to validate the tokens. The app should be available on v2.

Reply:
ADAL is needed here not to access the lab, but KeyVault. If we come up with a solution for accessing KeyVault using MSAL, it would be a big impact since Azure is pushing KeyVault strongly and a lot of developers experience AAD because of KV.

Reply:
msal c++ is using themselves to talk to the keyvault for test automation so it should be possible.  Travis was going to look into updating the callback token acquisition in the keyvault code to see if we can use msal there.  but that wno't be part of this PR so i'm going to close this comment stream out and we can discuss it in parallel.

---
In reply to: [251940425](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/838#discussion_r251940425) [](ancestors = 251940425)